fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/indexes/TextIndexTest.java

```diff
@@ -1039,7 +1043,167 @@ public void saveComplexWithAggressiveConflictRanges() throws Exception {
+            recordStore.saveRecord(doc1);
+            recordStore.saveRecord(doc2);
+            commit(context);
+        }
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context, hook);
+            Query.InvalidExpressionException err = assertThrows(Query.InvalidExpressionException.class,
+                    () -> recordStore.deleteRecordsWhere(SIMPLE_DOC, Query.field("text").equalsValue("foo bar")));
+            assertThat(err.getMessage(), containsString("deleteRecordsWhere not supported by index SimpleDocument$text"));
+        }
+    }
+
+    @Test
+    void deleteWhereComplexByGroup() {
+        final ComplexDocument zeroGroupDocument = ComplexDocument.newBuilder()
+                .setGroup(0)
+                .setDocId(1623L)
+                .setText(TextSamples.ROMEO_AND_JULIET_PROLOGUE)
+                .build();
+        final ComplexDocument oneGroupDocument = ComplexDocument.newBuilder()
+                .setGroup(1)
+                .setDocId(1623L)
+                .setText(TextSamples.ROMEO_AND_JULIET_PROLOGUE)
+                .build();
+        final RecordMetaDataHook hook = metaDataBuilder -> {
+            TextIndexTestUtils.addRecordTypePrefix(metaDataBuilder);
+            metaDataBuilder.addIndex(COMPLEX_DOC, COMPLEX_TEXT_BY_GROUP);
+        };
+
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context, hook);
+            recordStore.saveRecord(zeroGroupDocument);
+            recordStore.saveRecord(oneGroupDocument);
+            context.commit();
+        }
+
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context, hook);
+
+            RecordQuery query = RecordQuery.newBuilder()
+                    .setRecordType(COMPLEX_DOC)
+                    .setFilter(Query.and(
+                            Query.field("group").equalsParameter("group_value"),
+                            Query.field("text").text().containsPhrase("nought could remove")
+                    ))
+                    .build();
+            RecordQueryPlan plan = recordStore.planQuery(query);
+
+            // Both groups should have a single element before delete where
+            assertThat(plan.execute(recordStore, EvaluationContext.forBinding("group_value", 0)).asList().join(), hasSize(1));
+            assertThat(plan.execute(recordStore, EvaluationContext.forBinding("group_value", 1)).asList().join(), hasSize(1));
+
+            recordStore.deleteRecordsWhere(COMPLEX_DOC, Query.field("group").equalsValue(0));
+
+            // Group zero should be gone but group 1 should still have an element remaining
+            assertThat(plan.execute(recordStore, EvaluationContext.forBinding("group_value", 0)).asList().join(), empty());
+            assertThat(plan.execute(recordStore, EvaluationContext.forBinding("group_value", 1)).asList().join(), hasSize(1));
+        }
+    }
+
+    @Test
+    void deleteWhereGroupedMap() {
```

Comment:
This test also seems like it could use more text, and probably more entries in the map.

Reply:
Sure, added more documents and more map entries here. I also added a test that tries and delete with a predicate on the map key. It can't really do that because of the primary key on the record type, and it would require more work than I think it's worth to get a separate test working that aligned correctly, but it can at least assert that the error comes from the primary key (not the index)